test/e2e/cluster_upgrade.go

```diff
@@ -84,10 +84,9 @@ func ClusterUpgradeConformanceSpec(ctx context.Context, inputGetter func() Clust
 		By("Creating a workload cluster")
 
 		var controlPlaneMachineCount int64 = 1
-		// clusterTemplateWorkerMachineCount is used for ConfigCluster, as it is used for MachineDeployment and
-		// MachinePool, we actually get 2 * clusterTemplateWorkerMachineCount Machines
+		// clusterTemplateWorkerMachineCount is used for ConfigCluster, as it is used for MachineDeployments and
+		// MachinePools
 		var clusterTemplateWorkerMachineCount int64 = 2
-		var workerMachineCount = 2 * clusterTemplateWorkerMachineCount
```

Comment:
I was working on updating cluster-api-provider-packet to v1alpha4 and adopting the upstream test specs. The only thing preventing me from using ClusterUpgradeConformanceSpec is that it expects that a provider implements MachinePools, since we do not the spec always fails because the machine count does not match.